harmonica/tests/test_transformations.py

```diff
@@ -139,3 +235,159 @@ def test_derivative_upward_order2(sample_potential, sample_g_zz):
+    }
+    # need to drop upward coordinate (bug in xrft)
+    potential_padded = xrft.pad(
+        sample_potential.drop_vars("upward"),
+        pad_width=pad_width,
+    )
+    # Calculate second northing derivative and unpad it
+    second_deriv = derivative_northing(potential_padded, order=2)
+    second_deriv = xrft.unpad(second_deriv, pad_width)
+    # Compare against g_nn (trim the borders to ignore boundary effects)
+    trim = 6
+    second_deriv = second_deriv[trim:-trim, trim:-trim]
+    g_nn = sample_g_nn[trim:-trim, trim:-trim] * 1e-9  # convert to SI units
+    rms = root_mean_square_error(second_deriv, g_nn)
+    assert rms / np.abs(g_nn).max() < 0.1
+
+
+def test_laplace(sample_potential):
+    """
+    Test second order of derivative fullfill laplace equation
+    """
+    # Pad the potential field grid to improve accuracy
+    pad_width = {
+        "easting": sample_potential.easting.size // 3,
+        "northing": sample_potential.northing.size // 3,
+    }
+    # need to drop upward coordinate (bug in xrft)
+    potential_padded = xrft.pad(
+        sample_potential.drop_vars("upward"),
+        pad_width=pad_width,
+    )
+    # Calculate second northing derivative and unpad it
+    second_deriv_zz = derivative_upward(potential_padded, order=2)
+    second_deriv_ee = derivative_easting(potential_padded, order=2)
+    second_deriv_nn = derivative_northing(potential_padded, order=2)
+    second_deriv_zz = xrft.unpad(second_deriv_zz, pad_width)
+    second_deriv_ee = xrft.unpad(second_deriv_ee, pad_width)
+    second_deriv_nn = xrft.unpad(second_deriv_nn, pad_width)
+    # Compare g_nn + g_ee against -g_zz (trim the borders to ignore boundary
+    # effects)
+    trim = 6
+    second_deriv_sum = second_deriv_ee + second_deriv_nn
+    second_deriv_sum = second_deriv_sum[trim:-trim, trim:-trim]
+    second_deriv_zz = second_deriv_zz[trim:-trim, trim:-trim]
+    rms = root_mean_square_error(second_deriv_sum, -second_deriv_zz)
+    assert rms < 1e-20
+
+
+def test_upward_continuation(sample_g_z, sample_g_z_upward):
+    """
+    Test upward_continuation function against the synthetic model
+    """
+    # Pad the potential field grid to improve accuracy
+    pad_width = {
+        "easting": sample_g_z.easting.size // 3,
+        "northing": sample_g_z.northing.size // 3,
+    }
+    # need to drop upward coordinate (bug in xrft)
+    gravity_padded = xrft.pad(
+        sample_g_z.drop_vars("upward"),
+        pad_width=pad_width,
+    )
+    # Calculate upward continuation and unpad it
+    continuation = upward_continuation(gravity_padded, -10e3)
+    continuation = xrft.unpad(continuation, pad_width)
+    # Compare against g_z_upward (trim the borders to ignore boundary effects)
+    trim = 6
+    continuation = continuation[trim:-trim, trim:-trim]
+    g_z_upward = sample_g_z_upward[trim:-trim, trim:-trim]
+    rms = root_mean_square_error(continuation, g_z_upward)
+    assert rms / np.abs(g_z_upward).max() < 0.015
```

Comment:
Since the upward continuation is a low-pass filter, maybe we could also test this with `npt.assert_allclose()`. If not it's ok to leave it with the rms.

Reply:
Yup~